DEVELOPER_STARTING_GUIDE.md

```diff
@@ -0,0 +1,185 @@
+5. **Compare the old UI component to the newly adapted Shoelace component.** Refer to the [UI Component Library](https://component-library.dev.fe.union-investment.de/integration/storybook/index.html) to get an idea of what is being changed. This helps to see what is being lost or improved upon and should be summarized in the migration guide.
+6. **Keep the technical props and features of the Shoelace component that feel relevant.** It is easier to decide which ones can be removed at the end.
+7. **Convert the component styles to use a Tailwind first approach.** Shoelace does not use Tailwind and includes CSS files for every component. These should be carefully inspected and adapted. Work your way through the render() method of the component and remove CSS classes one by one, replacing them with a combination of Tailwind classes that implement our design. Keep in mind our designs may differ from the original styles applied by Shoelace, however, there are many small details (regarding `display`, `position`, or interaction states etc.) that should be carefully observed and adapted.
+8. **Implement any missing behavior.** Once the Shoelace component has been reduced and re-styled with Tailwind, add any missing functions that may be necessary to meet the component's requirements.
+9. **Write concise comments build up the auto-generated Storybook docs.** We rely primarily on the docs to communicate to the library users (developers). Write descriptive comments and organize properties semantically to the best of your ability.
+10. **Write concise Storybook stories to cover all visual scenarios.** See the Storybook section above for further information. All stories can furthermore be observed together in the docs.
+11. **Start with the jest tests from Shoelace.** Inside of `packages/components` you can test individual components with `pnpm test.component input` (where input is akin to the sd-input component). These will likely contain some misnamed properties or improper default values. Adapt them as needed to fit our use case.
+12. **Add any tests that are needed for new component behavior.** Motto: “Test the behavior, not the implementation.” Write tests that verify your components meet the expected requirements and specifications, ensuring they function correctly from the user's perspective. Don’t get too tied up trying to test all the technical details.
+13. **Author the full migration guide.** The migration guide should be placed in the `packages/components/src/docs/Migration` folder. The migration guide should be based on the migration guide template (`templates/migration-guide-template.mdx`) and be named by the old component name (e.g. ui-button.mdx).
+
+**Run `pnpm verify` in the root directory before creating a PR.** This will check that all formatting, tests, and build processes are working correctly to allow the pipeline to run successfully.
+
+### Styles
+
+If you are working on a style, follow the steps below:
+...
```

Comment:
This steps might have been skipped or missing.

Reply:
Right... just came to my mind, that this part needs to be defined. 

@Vahid1919 do you have any suggestions how to write/explain the steps?

Reply:
Yes. I will look through the styles I've done before and write some sort of draft.

Reply:
```suggestion
1. Create new styles in the `packages/components/src/styles` folder.
2. Set up a Storybook story alongside the styles, using a default story. This allows you to preview and iterate on the styles as you work.
3. **Each style should have an accompanying declaration file.** This is for documenting the style and its attributes. Follow the structure of our existing styles for guidance (eg: [sd-meta](https://github.com/solid-design-system/solid/blob/main/packages/components/src/styles/meta/meta.declaration.ts)).
4. Import your styles into the `packages/components/src/solid-styles.css` file. This step is crucial for enabling other Solid components to access the new styles.

_ You can always refer to the existing components and styles to familiarize yourself with the workflow!_
```

Reply:
@karlbaumhauer I think this covers it when it comes to making styles. Feel free to edit and reword as you see fit :)